error-prone-contrib/src/main/java/tech/picnic/errorprone/refasterrules/TimeRules.java

```diff
@@ -64,8 +65,47 @@ ZoneOffset after() {
     }
   }
 
+  /** Prefer more fluent way of converting an instant to a local date. */
+  static final class InstantToLocalDate {
+    @BeforeTemplate
+    LocalDate before(Instant instant, ZoneOffset zoneOffset) {
+      return LocalDate.ofInstant(instant, zoneOffset);
+    }
+
+    @AfterTemplate
+    LocalDate after(Instant instant, ZoneOffset zoneOffset) {
+      return instant.atOffset(zoneOffset).toLocalDate();
```

Comment:
While this is more "OOP-y" than the original, I'm not yet convinced that these rewrites are an improvement:
- The old code better conveys intent (IMHO).
- The old code seems more efficient.

So I wonder: shouldn't we define these rules in the opposite direction?

Reply:
Hi @Stephan202,
The four templates I added here is to extend the existing `InstantAtOffset` (which I renamed).
I don't have strong arguments about one way or another.

Cc @rickie

Reply:
I have no strong preference here to be honest. So let's do it the other way around then :).

Reply:
Doing it the other way around SGTM! (I think that the existing `OffsetDateTime.ofInstant(instant, zoneOffset)` -> `instant.atOffset(zoneOffset)` makes sense, but for the new cases the static factory method seems to be the way to go :+1:.)

Reply:
![can do](https://media0.giphy.com/media/DgLsbUL7SG3kI/giphy.gif?cid=ecf05e47cviotrmb0yn3kz8gk2oky3q1t0qpc3z169yi10jn&rid=giphy.gif&ct=g)